php/php.editor/tools/ASTPHP5Parser.cup

```diff
@@ -192,6 +192,28 @@ parser code {:
         return new ClassName(varleft, propertyListright, dispatch);
     }
 
+    Statement createAttributedStatement(Statement statement, List<Attribute> attributes) {
+        Statement attributedStatement = statement;
+        if (statement instanceof FunctionDeclaration) {
+            attributedStatement = FunctionDeclaration.create((FunctionDeclaration) statement, attributes);
+        } else if (statement instanceof ClassDeclaration) {
+            attributedStatement = ClassDeclaration.create((ClassDeclaration) statement, attributes);
+        } else if (statement instanceof InterfaceDeclaration) {
+            attributedStatement = InterfaceDeclaration.create((InterfaceDeclaration) statement, attributes);
+        } else if (statement instanceof TraitDeclaration) {
+            attributedStatement = TraitDeclaration.create((TraitDeclaration) statement, attributes);
+        } else if (statement instanceof FieldsDeclaration) {
+            attributedStatement = FieldsDeclaration.create((FieldsDeclaration) statement, attributes);
+        } else if (statement instanceof ConstantDeclaration) {
+            attributedStatement = ConstantDeclaration.create((ConstantDeclaration) statement, attributes);
+        } else if (statement instanceof MethodDeclaration) {
+            attributedStatement = MethodDeclaration.create((MethodDeclaration) statement, attributes);
+        } else {
+            assert false;
+        }
+        return attributedStatement;
+    }
```

Comment:
I've added the static create methods to each node to add attributes (recreate the node with attributes).